Introduce CAPTCHA Requirements to the registration

SaadAli100 opened the pull request:
Register your website for reCAPTCHA by visiting the Google reCAPTCHA website. Decide whether to utilize reCAPTCHA v2 or v3, depending on your preference. You will receive a site key and a secret key upon registration/login.

add the reCAPTCHA widget to your form. Place this inside your registration/login form where you want the CAPTCHA to appear: data-sitekey="YOUR_SITE_KEY"


Verify the server-side reCAPTCHA response:
Google reCAPTCHA sends you a response once the form is submitted, which you should check on your server. Utilizing your secret key and response token, send a request to Google via your server-side script.

# PR Template
# Any italic text should be deleted from the final Pull Request text, including this line
## Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Testing Checklist
- [ ] Tested in latest Chrome
- [ ] Tested in latest Safari
- [ ] Tested in latest Firefox

## Checklist
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have requested a review from NAMES HERE on the Pull Request

## Notes for PRs 
Below the commit message field you decide whether you are committing your changes to the current branch or a new branch. If the current brain is the main branch (develop), commit to a new branch.

### Filling In CheckBox Example
- [x] Example Filled
- [ ] Example Empty

SaadAli100 committed:
Introduce CAPTCHA Requirements to the registration 

Register your website for reCAPTCHA by visiting the Google reCAPTCHA website. Decide whether to utilize reCAPTCHA v2 or v3, depending on your preference. You will receive a site key and a secret key upon registration/login.

add the reCAPTCHA widget to your form. Place this inside your registration/login form where you want the CAPTCHA to appear: data-sitekey="YOUR_SITE_KEY"


Verify the server-side reCAPTCHA response:
Google reCAPTCHA sends you a response once the form is submitted, which you should check on your server. Utilizing your secret key and response token, send a request to Google via your server-side script.

Signed-off-by: SaadAli100 <150887259+SaadAli100@users.noreply.github.com>

hmm007 reviewed (approved):
![image](https://github.com/DataBytes-Organisation/Neo_Dolfin/assets/112069519/d8c32611-f228-445e-8224-2ccb63d6504e)


This one works and is flexible - proper implementation will be enabled once Dolfin has a site key . I added <! ---> to make it a comment because currently Dolfin does not have a site key, and so we don't need it to be sitting there redundant.

hmm007 committed:
Hiding CAPTCHA Requirement

Just added <!-- to line 85 and --> to line 88. 

This is so that the CAPTCHA requirement box isn't there until Dolfin gets its own site key.

Signed-off-by: hmm007 <112069519+hmm007@users.noreply.github.com>

hmm007 commented:
I added <!-- to line 85 and --> to line 88 so that the CAPTCHA can be re-implemented when Dolfin has a site key.
